### ISSUE TITLE
Create a build config to mix original BASIC v2 ROM with open kernal?

### ISSUE THREAD

**Issue by ssuukk** [gardners/open-roms#73]
Implementing the whole BASIC might still take some time. Would it be possible to create a build config that would instead build only the kernal and glue the smaller part to existing BASIC v2 ROM?

**Comment by FeralChild64** [comment]
You can already make such ROMs for Commodore 64 and Ultimate 64, try `make test_hybrid` or `make test_hybrid_u64`. But you won’t be able to legally distribute such ROMs due to licensing issues (Kernal actually contains some portions of BASIC). You will lose some features, though, like the DOS Wedge.

For MEGA65 native build, such a ROM is IMHO a bad idea - you really want the GO64 command to work :) Also, the only challenging part of BASIC 2.0 left  to implement is a floating point library (it’s partially done):
- tokenizer and general execution engine are working
- expression parser is there
- string variables are there, including the garbage collector

Commands and functions themselves are typically easy to implement.

**Comment by ssuukk** [comment]
OK, honestly I don't give a duck about legally distributing and so on. I'd be more than happy to copy paste original ROM code to fill in the gaps. Is there any piece of documentation how to do it in a least painful way?

**Comment by gcasa** [comment]
> OK, honestly I don't give a duck about legally distributing and so on. I'd be more than happy to copy paste original ROM code to fill in the gaps. Is there any piece of documentation how to do it in a least painful way?

Sorry if I am interfering in this conversation but I feel obliged to say that you should give a "duck" because when all is said and done, it's better to have a completely free (legally) ROM than it is to have a legally compromised one.

**Comment by ssuukk** [comment]
Well, even if you reported me to copyright police, I still doubt  they would care that in the cooolnes of my home I infriged a copyright (that nobody cares about) of a 50-year ROM (than nobody cares about). So, no. It isn't better, just goofier.

**Comment by gcasa** [comment]
> Well, even if you reported me to copyright police, I still doubt  they would care that in the cooolnes of my home I infriged a copyright (that nobody cares about) of a 50-year ROM (than nobody cares about). So, no. It isn't better, just goofier.

I have no intention of reporting you, don't be silly.  You'll find that, when it comes to such things there is always SOMEONE who does care. Generally, you don't have much to worry about... you're right.  But it is always better to be safe than sorry in my opinion.

All of that being said it is likely okay to consider the original ROM as "inspiration". :) In the spirit of what Franklin did with the Apple ROM.  In the long run it's the better way to go.  

For instance, I am the lead of a project known as GNUstep which is an implementation of the Cocoa (macOS) APIs.  At one point a dev on my team copied a few headers DIRECTLY.  While you could make the argument that "no one would care" that falls pretty flat when someone takes you to court over it which is a risk if you copy.  

I handled the situation by removing and rewriting the header.  It wasn't hard.  If you have any talent as a developer it should be easy to rewrite it rather than copy it. 

At any rate.  Up to you.  But, like I said, better safe than sorry.

**Comment by gcasa** [comment]
> Well, even if you reported me to copyright police, I still doubt  they would care that in the cooolnes of my home I infriged a copyright (that nobody cares about) of a 50-year ROM (than nobody cares about). So, no. It isn't better, just goofier.

Oh and it is MUCH goofier to infringe someone else's work and dismiss it as not being important.  Much MUCH goofier because you're needlessly risking exposing both yourself and the work done by others to legal repercussions.

**Comment by ssuukk** [comment]
Don't worry, I'll delete all tracking cookies on my C64 with this naughty ROM. No Commodore lawyer will notice anything!

**Comment by gcasa** [comment]
> Don't worry, I'll delete all tracking cookies on my C64 with this naughty ROM. No Commodore lawyer will notice anything!

Except maybe this thread. Suit yourself. 😂 Whatever

**Comment by gcasa** [comment]
> Don't worry, I'll delete all tracking cookies on my C64 with this naughty ROM. No Commodore lawyer will notice anything!

If you were part of my project I would not allow you to contribute legally questionable code back into the project.  Speaking only for myself, of course.  But, I'm out.  I only intended to make a quick comment and it has become more than that.

Copyright laws are important whether you happen to think they are or not.  If you're doing it for yourself that's one thing, have at it... gleefully.  If you're contributing it to a public project that is an entirely different affair.  But, as I said, not my project... not my concern.  In the US I have learned to be more careful about such things. Over and out. GC
